server/schedule/checker_controller.go

```diff
@@ -70,11 +73,26 @@ func (c *CheckerController) CheckRegion(region *core.RegionInfo) []*operator.Ope
 
 	if c.opts.IsPlacementRulesEnabled() {
 		if op := c.ruleChecker.Check(region); op != nil {
+			added := false
+			// ignore spilt or merge
+			if op.Kind()&operator.OpReplica != 0 {
+				missPeers, count := c.GetRuleChecker().GetMissPeer(region)
+				// miss replicate more than majority should has high priority.
+				// if less than majority, it will add waiting
+				if missPeers > count/2 {
+					c.AddMissRegions(missPeers, region)
+					added = true
+				} else {
+					c.RemoveMissRegions([]uint64{region.GetID()})
+				}
+			}
 			if opController.OperatorCount(operator.OpReplica) < c.opts.GetReplicaScheduleLimit() {
 				return []*operator.Operator{op}
 			}
 			operator.OperatorLimitCounter.WithLabelValues(c.ruleChecker.GetType(), operator.OpReplica.String()).Inc()
-			c.regionWaitingList.Put(region.GetID(), nil)
+			if !added {
```

Comment:
What if the region has 2 peers and the max replica is 3?

Reply:
the region will work continue ,so it will be putted to waiting queue

Reply:
Yes, but I am afraid we have too many data structures like list/queue .etc which may not easy to maintain in future development. Perhaps we can unify them someday.